hw/top_earlgrey/data/tb__xbar_connect.sv.tpl

```diff
@@ -41,6 +41,12 @@ for xbar in top["xbar"]:
      force dut.top_earlgrey.u_``inst_name``.``sig_name``_o = ``tl_name``_tl_if.d2h; \
      uvm_config_db#(virtual tl_if)::set(null, $sformatf("*%0s*", `"tl_name`"), "vif", \
                                         ``tl_name``_tl_if);
+
+`define DRIVE_TL_EXT_DEVICE_IF(tl_name, port_name) \
```

Comment:
@weicaiyang please take a look. Not really sure if this is the fix you prefer, but hopefully this will get us past private CI for now.

Reply:
as @sriyerg suggested, we should connect to the IP port rather interconnect. so it will be dut.ast_wrapper.bus_i/o
can just change DRIVE_TL_DEVICE_IF to accept another hierarchy. 

BTW, it will be easier to use `u_ast_wrapper` as instance name, rather than `ast_wrapper` 

````
`define DRIVE_TL_DEVICE_IF(tl_name, inst_name, sig_name, path = dut.top_earlgrey)
     force ``tl_name``_tl_if.h2d = ``path``.u_``inst_name``.``sig_name``_i; \
     force ``path``.u_``inst_name``.``sig_name``_o = ``tl_name``_tl_if.d2h; \
     force dut.top_earlgrey.u_``inst_name``.``sig_name``_o = ``tl_name``_tl_if.d2h; \
     uvm_config_db#(virtual tl_if)::set(null, $sformatf("*%0s*", `"tl_name`"), "vif", \
     uvm_config_db#(virtual tl_if)::set(null, $sformatf("*%0s*", `"tl_name`"), "vif", \
                                        ``tl_name``_tl_if);	                                        ``tl_name``_tl_if);
```